src/Raven.Server/Documents/Patch/PatchDocument.cs

```diff
@@ -267,9 +268,17 @@ protected PatcherOperationScope ApplySingleScript(DocumentsOperationContext cont
             }
         }
 
-        protected void SetupInputs(Document document, PatcherOperationScope scope, Engine jintEngine)
+        protected void SetupInputs(Document document, PatcherOperationScope scope, Engine jintEngine, string docId)
         {
-            jintEngine.SetValue("__document_id", document.Key);
+            jintEngine.SetValue("__document_id", docId);
+
+            if (document == null)
```

Comment:
I think that you should move it to PatchDocument.Apply and pass this document to the ApplySingleScript from there.